src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
The line at https://github.com/realm/realm-object-store/pull/363/files#diff-f3165f1bbea4d4c8abf9c03a4b4e86f9R509, on the other hand, is crucial.